File: MatriX/samples/vb/Weather/WeatherIq.vb
Imports Matrix.Xmpp.Client

Public Class WeatherIq
	Inherits Iq
	Public Sub New()
		GenerateId()
	End Sub

	Public Property Weather() As Weather
		Get
			Return Element(Of Weather)()
		End Get
		Set
			Replace(value)
		End Set
	End Property
End Class

File: MatriX/samples/csharp/Weather/WeatherIq.cs
using Matrix.Xmpp.Client;

namespace Weather
{
    public class WeatherIq : Iq
    {
        public WeatherIq()
        {
            GenerateId();
        }

        public Weather Weather
        {
            get { return Element<Weather>(); }
            set { Replace(value); }
        }
    }
}